sdk/eventhubs/azure-messaging-eventhubs/TROUBLESHOOTING.md

```diff
@@ -0,0 +1,269 @@
+
+### SSL handshake failures
+
+This error can occur when the proxy is not configured properly.  Ask the customer to test their connection to Event Hubs without using a proxy.
+
+### Socket exhaustion errors
+
+Customers should keep an instance of their Event Hub client as long as possible because by default, the creation of a each client results in creating an AMQP connection which uses a socket.  Additionally, clients inherit from `java.io.Closeable`, so they should be calling `close()` when they are finished.
+
+To use the same AMQP connection when creating multiple clients, customers can use the `EventHubClientBuilder.shareConnection()` flag, hold a reference to that `EventHubClientBuilder`, and create new clients from that same builder instance.
+
+### Connect using an IoT connection string
+
+This is not a supported scenario in our current library.  The sample describes how to acquire an IoT connection string which can then be used to instantiate the Event Hub clients: [IoTConnectionString.java][IoTConnectionString]
+
+Further reading:
+* https://docs.microsoft.com/azure/iot-hub/iot-hub-dev-guide-sas#security-tokens
+* https://docs.microsoft.com/samples/azure/azure-sdk-for-net/iothub-connect-to-eventhubs
+
+### Cannot add "TransportType=AmqpWebSockets" to the connection string
+
+The legacy clients allowed customers to modify the connection string to enable capabilities.  In this case, use web sockets to connect to Event Hubs.  To achieve the same scenario, see [PublishEventsWithSocketsAndProxy.java][PublishEventsWithWebSocketsAndProxy].
+
+### Cannot add "Authentication=Managed Identity" to the connection string
+
+The legacy clients allowed customers to modify the connection string to enable capabilities.  In this case, connect to Event Hubs using managed identity rather than a connection string.  To achieve the same scenario, see [PublishEventsWithAzureIdentity.java][PublishEventsWithAzureIdentity].
+
+For more information about our identity library see: https://devblogs.microsoft.com/azure-sdk/authentication-and-the-azure-sdk/
+
+## Enable and configure logging
+
+Azure SDK for Java offers a consistent logging story to help aid in troubleshooting application errors and expedite their resolution. The logs produced will capture the flow of an application before reaching the terminal state to help locate the root issue. View the [logging][Logging] wiki for guidance about enabling logging.
+
+In addition to enabling logging, setting the log level to `VERBOSE` or `DEBUG` provides insights into the library's state.  Below are sample log4j2 and logback configurations to reduce excessive
+log messages when verbose logging is enabled.
+
+### Configuring Log4J 2
+
+1. Add the following dependencies in your pom.xml.
+    ```xml
+    <dependency>
+        <groupId>org.apache.logging.log4j</groupId>
+        <artifactId>log4j-api</artifactId>
+        <version>2.14.1</version>
+    </dependency>
```

Comment:
Do we need a way to keep these versions up-to-date given the number of CVEs Log4j has seen recently so that we don't recommend customers to use vulnerable libraries for logging?